lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/client/uploader/JobUploader.java

```diff
@@ -39,8 +43,10 @@
 
   @Inject
   JobUploader(final YarnConfiguration yarnConfiguration,
-              final JobSubmissionDirectoryProvider jobSubmissionDirectoryProvider) throws IOException {
+              final JobSubmissionDirectoryProvider jobSubmissionDirectoryProvider,
+              @Parameter(FileSystemUrl.class) final String fileSystemUrl) throws IOException {
     this.jobSubmissionDirectoryProvider = jobSubmissionDirectoryProvider;
+    YarnConfigurationSetUp.setFileSystemUrl(yarnConfiguration, fileSystemUrl);
```

Comment:
So we pass YARN configuration and a filesystem URL into the `JobUploader`. I can understand that this is needed to upload files for the YARN job. What I *don't* like is that `JobUploader` actually modifies YARN configuration as a side effect. We should avoid such things even if they are harmless (like, I hope it is in this particular case).

What we should do instead is to identify *one* place where we build our `YarnConfiguration` instance (say, in Tang legacy constructor), and modify it *there*. It will lead to much fewer changes in other parts of the code.